### PR TITLE
Use updates stable relations from js-sdk

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/2412

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/structures/RoomView.tsx [hunk 1]
```diff
@@ -974,9 +974,8 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
 
         CHAT_EFFECTS.forEach(effect => {
             if (containsEmoji(ev.getContent(), effect.emojis) || ev.getContent().msgtype === effect.msgType) {
-                // For initial threads launch, chat effects are disabled
-                // see #19731
-                if (!SettingsStore.getValue("feature_thread") || !ev.isThreadRelation) {
+                // For initial threads launch, chat effects are disabled see #19731
+                if (!SettingsStore.getValue("feature_thread") || !ev.isRelation(THREAD_RELATION_TYPE.name)) {
```
Comment (turt2live) [inline src/components/structures/RoomView.tsx:978]: we should be checking both `name` and `altName`, just give up on the namespace entirely, or document why we don't care about the `altName` with code comments.

File: src/utils/EventUtils.ts [hunk 2]
```diff
@@ -229,7 +230,10 @@ export async function fetchInitialEvent(
         initialEvent = null;
     }
 
-    if (initialEvent?.isThreadRelation && client.supportsExperimentalThreads() && !initialEvent.getThread()) {
+    if (client.supportsExperimentalThreads() &&
+        initialEvent?.isRelation(THREAD_RELATION_TYPE.name) &&
```
Comment (turt2live) [inline src/utils/EventUtils.ts:234]: same thing about `altName`

**Review by turt2live — CHANGES_REQUESTED** [review]
generally seems fine, though the purpose of a `NamespacedValue` appears to be diminished/lost here.

File: src/components/structures/RoomView.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/RoomView.tsx:978]: `THREAD_RELATION_TYPE` is a `ServerControlledNamespacedValue` to `name` automagically changes between `name` and `altName` of the underlying namespaces value, for things like Threads we only want to handle one or the other depending on which your server understands

File: src/utils/EventUtils.ts (hunk 2 above)
Comment (t3chguy) [reply src/utils/EventUtils.ts:234]: Same as above

File: src/components/structures/RoomView.tsx (hunk 1 above)
Comment (turt2live) [reply src/components/structures/RoomView.tsx:978]: ... that shouldn't be a NamespacedValue then, as it's breaking the contract one can expect of a namespaced value. Not worth fixing in this PR, but also not exactly happy about that being a thing.

**Review by turt2live — APPROVED** [review]

**Pull request merged**